staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -300,3 +317,22 @@ func (le *LeaderElector) maybeReportTransition() {
 		go le.config.Callbacks.OnNewLeader(le.reportedLeader)
 	}
 }
+
+// Check will determine if the current lease is expired by more than timeout.
```

Comment:
maybe calling it "maxTolerableExpiredLease" or something would be clearer?

Alternatively, you could make a function that just returns the lease age and folks could do whatever they wanted.

Reply:
Would rather not expose more internals than are needed for the use case.